CMSgov/design-system
# [WNMGDS-526] Optimize scripts

### Summary
After adding support for typescript I found our build scripts to be incredibly slow. This PR is to optimize some of these scripts and reorganize the code to be easier to understand. 

### How to test
First ensure a clean build
- run `git clean -fdx` on the main design system branch
- run `yarn build` on the main design system branch
- run `yarn start` on the child design system 

Test all the build scripts on the core CMSDS and the WIP m.gov child DS
[mgov.zip](https://github.com/CMSgov/design-system/files/4886099/mgov.zip)
In particular, test `yarn start` with the mgov child DS
- Updating React props / Typescript types
- Updating component code will compile code and site
- Updating styles will update site
- Updating example files will update site
